### ISSUE TITLE
Alternatives

### ISSUE THREAD

**Issue by hax** [js-choi/proposal-method-extraction#1]
Except a special binary notation for method extraction (`obj&.method` or `obj.&method()`), there are some alternatives:

## partial func application proposal

`obj.method(...)` (syntax precedents: PHP!)

Note if we go with Hack style pipeline, we will not have partial func app anymore. But it's very possible to have `+> obj.method(...)` instead.

## refs proposal: 
see https://github.com/rbuckton/proposal-refs/issues/9

The problem is, as current ref proposal we actually need code like `setTimeout(do { let ref m = ref obj.method; m }` which seems doesn't make sense...

## Extensions proposal:

Though extensions do not support method extraction directly, it allow userland solution with nice syntax like `obj::bindMethod:method`, and it could also support extract accessors.

**Comment by js-choi** [comment]
Yes, sorry for the confusion—I wrote this proposal as an experiment. But I’m planning to possibly shelve it, in favor of [possibly resurrecting the bind operator](https://github.com/js-choi/proposal-bind-operator).

More specifically, I want to address @ljharb’s security use case: protecting against mutation of `Function.prototype.call` and `bind` by an adversary. (From what I can tell, that use case is out of scope of your Extensions proposal.) 

By the way, I’ve already been planning to reach out to you later to talk more about [js-choi/proposal-bind-operator](https://github.com/js-choi/proposal-bind-operator) versus Extensions. I’ll reach out to you on Matrix.

Sorry again for the confusion!

**Comment by hax** [comment]
I think this is a good proposal, I create this issue just for complement :)
